### PR TITLE
Replace gulp-util usages

### LINKED ISSUE #180: Dependency deprecation warning: gulp-util (npm)
On registry `https://registry.npmjs.org/`, the "latest" version (v3.0.8) of dependency `gulp-util` has the following deprecation notice:

`gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5`

Marking the latest version of an npm package as deprecated results in the entire package being considered deprecated, so contact the package author you think this is a mistake.

Affected package file(s): `package.json`

If you don't care about this, you can close this issue and not be warned about `gulp-util`'s deprecation again. If you would like to completely disable all future deprecation warnings then add the following to your config:

```
"suppressNotifications": ["deprecationWarningIssues"]
```

### PR DESCRIPTION
**Pull request by bdukes**
Fixes #180

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,12 +28,18 @@
     "release": "standard-version"
   },
   "dependencies": {
-    "gulp-util": "^3.0.8",
-    "through2": "^2.0.3"
+    "@types/ansi-colors": "^3.2.0",
+    "@types/fancy-log": "^1.3.1",
+    "@types/plugin-error": "^0.1.1",
+    "@types/vinyl": "^2.0.3",
```
Comment (ikatyang) [inline package.json]: These `@types` should be `devDependencies`, we did not expose them to the user.

**Comment by ikatyang** [comment]
It seems we should put the original `gulpfile.ts` somewhere so that we could use it in the gulp 3 build, or should we simply drop it? I haven't used it for a while so I'm not sure how's the status now.

**Comment by bdukes** [comment]
I think it's still worth testing with Gulp 3.  I've added some logic to switch out the Gulpfile, and now all builds are passing 🎉

Thanks for your help, let me know if there's anything else you want from me on this PR!

**Comment by ikatyang** [comment]
Thanks!

**Pull request merged**